app/questionnaire/routing/helpers.py

```diff
@@ -0,0 +1,30 @@
+from datetime import datetime
+from decimal import Decimal
+from functools import wraps
+from typing import Any, Callable, Union
+
+ValueTypes = Union[bool, str, int, float, Decimal, None, datetime]
+
+
+def _casefold(value: Union[list, ValueTypes]) -> Union[list, ValueTypes]:
+    if isinstance(value, str):
+        return value.casefold()
+
+    if isinstance(value, list):
+        return [_casefold(v) for v in value]
+
+    return value
+
+
+def casefold(func: Callable) -> Any:
+    @wraps(func)
+    def wrapper(*args: Any, **kwargs: Any) -> Any:
+        casefolded_args = [_casefold(arg) for arg in args]
+        casefolded_kwargs = {k: _casefold(v) for k, v in kwargs.items()}
+        return func(*casefolded_args, **casefolded_kwargs)
+
+    return wrapper
+
+
+def datetime_as_midnight(date: datetime) -> datetime:
```

Comment:
Never a massive fan of the term midnight as can be ambiguous to some (today/tomorrow) although not pretty datetime_as_today_start at least removes some confusion

Reply:
See what others think, as I'm not a big fan of `datetime_as_today_start`. I thought midnight was clear since it for a given datetime. Happy to consider other schemes if it is prone to tripping people up.

Reply:
No neither do I, I just dislike the term midnight. Straw poll of the one person around me reckoned midnight was tomorrow after closing one eye and screwing up nose.

Reply:
@ajmaddaford got a preference or any other suggestions?

Reply:
`datetime_at_start_of_day`?

Reply:
This method has now been dropped because `date` is now returned instead of `datetime`.

Reply:
Resolving this as the method has been removed.